test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,250 @@
+    });
+
+    it('deleteMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.deleteMany({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateOne(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('updateMany returns queryPlanner explain result', async function () {
+      const explanation = await collection.updateMany(
+        { a: 1 },
+        { $inc: { a: 2 } },
+        { explain: true }
+      );
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('distinct returns queryPlanner explain result', async function () {
+      const explanation = await collection.distinct('a', {}, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('findOneAndDelete returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOneAndDelete({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('allPlansExecution returns verbose queryPlanner explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+      expect(explanation).nested.property('executionStats.allPlansExecution').to.exist;
+    });
+
+    it('findOne returns queryPlanner explain result', async function () {
+      const explanation = await collection.findOne({ a: 1 }, { explain: true });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('find returns queryPlanner explain result', async () => {
+      const [explanation] = await collection.find({ a: 1 }, { explain: true }).toArray();
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+  });
+
+  context('when explain is set to false', () => {
+    it('only queryPlanner property is used in explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: false });
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+
+    it('find returns "queryPlanner" explain result specified on cursor', async function () {
+      const explanation = await collection.find({ a: 1 }).explain(false);
+      expect(explanation).property('queryPlanner').to.exist;
+    });
+  });
+
+  context('when explain is set to "queryPlanner"', () => {
+    it('only queryPlanner property is used in explain result', async function () {
+      const explanation = await collection.deleteOne({ a: 1 }, { explain: 'queryPlanner' });
+      expect(explanation).property('queryPlanner').to.exist;
```

Comment:
this assertion by itself does not guarantee that *only* `queryPlanner` exists; this test case would pass even if other properties were present on the result